File: python/15c_271A_BeautifulYear.py
def main():
    n = int(input())
    for i in range(n + 1, 9013):
        if distinct(i):
            print(i)
            break

def distinct(year):
    hmap = {}
    while year > 0:
        check = year % 10
        if check in hmap:
            return False
        year //= 10
        hmap[check] = 1
    return True

if __name__ == "__main__":
    main()

File: cpp/15c_271A_BeautifulYear.cpp
#include<bits/stdc++.h>

bool distinct(int year){
    std::unordered_map<int,int> hmap;
    int temp = year;
    while ( temp > 0){
        int check = temp % 10;
        temp /= 10;
        if(hmap.find(check) != hmap.end()){
            return false;
        }
        else{
            hmap[check]++;
        }
    }
    return true;
}

int main(){
    int y;
    std::cin>>y;
    for(int i = y + 1; i < 9013; i++){
        if (distinct(i)){
            std::cout<<i;
            break;
        }
    }
    return 0;
}